pkg/kncloudevents/message_sender_new_test.go

```diff
@@ -1,524 +0,0 @@
-	require.NoError(t, err)
-
-	got, err := request.SendWithRetries(config)
-
-	require.Equal(t, 5, int(atomic.LoadInt32(&n)))
-	require.NoError(t, err)
-	require.Equal(t, http.StatusOK, got.StatusCode)
-}
-
-func TestCloudEventRequest_SendWithRetriesOnNetworkErrors(t *testing.T) {
-
-	n := int32(10)
-	linear := v1.BackoffPolicyLinear
-	target := "127.0.0.1:63468"
-
-	calls := make(chan struct{})
-	defer close(calls)
-
-	nCalls := int32(0)
-
-	cont := make(chan struct{})
-	defer close(cont)
-
-	go func() {
-		for range calls {
-
-			nCalls++
-			// Simulate that the target service is back up.
-			//
-			// First n/2-1 calls we get connection refused since there is no server running.
-			// Now we start a server that responds with a retryable error, so we expect that
-			// the client continues to retry for a different reason.
-			//
-			// The last time we return 200, so we don't expect a new retry.
-			if n/2 == nCalls {
-
-				l, err := net.Listen("tcp", target)
-				assert.Nil(t, err)
-
-				s := httptest.NewUnstartedServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
-					if n-1 != nCalls {
-						writer.WriteHeader(http.StatusServiceUnavailable)
-						return
-					}
-				}))
-				defer s.Close() //nolint // defers in this range loop won't run unless the channel gets closed
-
-				assert.Nil(t, s.Listener.Close())
-
-				s.Listener = l
-
-				s.Start()
-			}
-			cont <- struct{}{}
-		}
-	}()
-
-	r, err := RetryConfigFromDeliverySpec(v1.DeliverySpec{
-		Retry:         pointer.Int32(n),
-		BackoffPolicy: &linear,
-		BackoffDelay:  pointer.String("PT0.1S"),
-	})
-	assert.Nil(t, err)
-
-	checkRetry := r.CheckRetry
-
-	r.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
-		calls <- struct{}{}
-		<-cont
-
-		return checkRetry(ctx, resp, err)
-	}
-
-	targetAddressable := urlstrToAddressable(t, "http://"+target)
-	request, err := NewCloudEventRequest(context.TODO(), targetAddressable)
-	assert.Nil(t, err)
-
-	_, err = request.SendWithRetries(&r)
-	assert.Nil(t, err)
-
-	// nCalls keeps track of how many times a call to check retry occurs.
-	// Since the number of request are n + 1 and the last one is successful the expected number of calls are n.
-	assert.Equal(t, n, nCalls, "expected %d got %d", n, nCalls)
-}
-
-/*
- * Test TestGenerateBackoffFnWithRetryAfter
- *
- * This test validates that generateBackoffFn() is correctly enforcing
- * the Retry-After headers based on RetryConfig.RetryAfterMaxDuration.
- * This is tested directly, rather than indirectly via SendWithRetries(),
- * due to the complexity in ensuring timing checks of sequential requests.
- * The CI pipeline is not reliable enough to consistently ensure any
- * specific response times, making the tests inherently flaky during slow
- * test runs. This version is also much faster to execute as it doesn't
- * to stand-up test servers.
- */
-func TestGenerateBackoffFnWithRetryAfter(t *testing.T) {
```

Comment:
@creydr since we are moving the `generateBackoffFn()` function, should we also keep move this test?

Reply:
Uh yes. Good catch :+1: Will add it again

Reply:
Added in d23f0f9d8cd6f20ca8a689bd0ab7f41d074e023a again